AdminCore.DAL/Entity Framework/EntityFrameworkRepository.cs

```diff
@@ -76,17 +93,27 @@ public virtual void Update(T entityToUpdate)
       ((EntityFrameworkContext)_context).Entry(entityToUpdate).State = EntityState.Modified;
     }
 
-    public IQueryable<T> IncludeEntities(IQueryable<T> query, Expression<Func<T, object>>[] includeProperties)
+    private static IQueryable<T> IncludeEntities(IQueryable<T> query, 
+      params (Expression<Func<T, object>> includeProperty, Expression<Func<object, object>>[] thenIncludes)[] includeDatas)
     {
-      if (includeProperties != null)
-      {
-        foreach (var includeProperty in includeProperties)
+      foreach (var (includeProperty, thenIncludes) in includeDatas)
+      { 
+        if (includeProperty != null)
         {
-          query = query.Include(includeProperty);
+          var newQuery = query.Include(includeProperty);
+          if (thenIncludes != null && thenIncludes.Any())
```

Comment:
this condition could be simply `thenIncludes?.Any() == true`

Reply:
You can't quite do ```thenIncludes?.Any()``` but you can do ```thenIncludes?.Any() ?? false```.

I'm not sure which one reads better, maybe the original if statement?